File: app/src/main/java/com/kotobyte/base/BaseApplication.java
package com.kotobyte.base;

import android.app.Application;

import com.kotobyte.models.db.DictionaryProvider;
import com.kotobyte.utils.DefaultConfiguration;


public class BaseApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();

        ServiceLocator services = ServiceLocator.getInstance();
        services.mConfiguration = new DefaultConfiguration(this);
        services.mDatabaseProvider = new DictionaryProvider(services.mConfiguration, getAssets());
    }
}

File: app/src/main/kotlin/com/kotobyte/base/BaseApplication.kt
package com.kotobyte.base

import android.app.Application

import com.kotobyte.models.db.DictionaryProvider
import com.kotobyte.utils.DefaultConfiguration


class BaseApplication : Application() {

    override fun onCreate() {
        super.onCreate()

        ServiceLocator.configuration = DefaultConfiguration(this)
        ServiceLocator.databaseProvider = DictionaryProvider(ServiceLocator.configuration!!, assets)
    }
}
